cms/src/hooks/sync-to-legacy/filters/episodes.ts

```diff
@@ -0,0 +1,143 @@
+    console.log('episode created!');
+    console.log(m);
+    // get legacy id
+    let asset: any
+    if (p.asset_id) {
+        asset = (await c.database("assets").select("*").where("id", p.asset_id))[0];
+    }
+    let image = null
+    if (p.image_file_id != null) {
+        image = (await c.database("directus_files").select("*").where("id", p.image_file_id))[0];
+    }
+    console.log("directus", p)
+
+    // update it in original 
+    let patch: Partial<EpisodeEntity> = {
+        VideoId: asset?.legacy_id ?? 1041, // TODO: remove this temp id
+        Published: p.publish_date as unknown as Date,
+        AvailableTo: p.available_to as unknown as Date,
+        AvailableFrom: p.available_from as unknown as Date,
+        Status: getStatusFromNew(p.status),
+        LastUpdate: p.date_created as unknown as Date ?? new Date(),
+        Visibility: 1,
+        AllowSpecialAccess: false,
+        AllowSpecialAccessFKTB: false,
+    }
+    patch.TitleId = await createLocalizable(oldKnex)
+    patch.DescriptionId = await createLocalizable(oldKnex)
+    patch.LongDescriptionId = await createLocalizable(oldKnex)
+    p.legacy_title_id = patch.TitleId
+    p.legacy_description_id = patch.DescriptionId
+    p.legacy_extra_description_id = patch.LongDescriptionId
+    
+    if (image != null) {
+        patch.Image = "https://brunstadtv.imgix.net/"+image.filename_disk
+    } 
+
+    if (p.status == "published") {
+        patch.Status = 1
+    } else {
+        patch.Status = 0
+    }
+    
+    if (p.type === "episode" || p.season_id) {
+        let season = (await c.database("seasons").select("*").where("id", p.season_id))[0];
+        patch.SeasonId = season.legacy_id
+        patch.EpisodeNo = p.episode_number
+        let legacyEpisode = await oldKnex<EpisodeEntity>("episode").insert(patch).returning("*")
+        p.legacy_id = legacyEpisode[0].Id
+        console.log(legacyEpisode)
+    } else if (p.type === "standalone") {
+        let legacyProgram = await oldKnex<EpisodeEntity>("program").insert(patch).returning("*")
+        console.log(legacyProgram)
+        p.legacy_program_id = legacyProgram[0].Id
+    }
+
+    console.log("insert", patch)
+}
+
+export async function updateEpisode(p, m, c) {
+    console.log('Item updated!');
+    console.log(m);
+    if (m.collection != "episodes") {
+        return
+    }
+    // get legacy id
+    let epBeforeUpdate = (await c.database("episodes").select("*").where("id", m.keys[0]))[0];
+
+    // update it in original 
+    let patch: Partial<EpisodeEntity> = {
+        Published: p.publish_date as unknown as Date,
+        AvailableTo: p.available_to as unknown as Date,
+        AvailableFrom: p.available_from as unknown as Date,
+        LastUpdate: new Date()
+    }
+    if (p.asset_id) {
+        let asset = (await c.database("assets").select("*").where("id", p.asset_id))[0];
+        patch.VideoId = asset.legacy_id
+    } else if (p.asset_id === null) {
+        patch.VideoId = null
+    }
```

Comment:
Is it intentional that you check that strictly for `null` or should it just be a plain else (not sure what the implications are)

Reply:
intentional. This is an update filter, so if its "undefined" it wasnt changed, but if its "null" we know directus explicitly set it to null.